app/plugin_api_test.go

```diff
@@ -556,6 +556,36 @@ func TestPluginAPIUserCustomStatus(t *testing.T) {
 	userCs, err = th.App.GetCustomStatus(user1.Id)
 	assert.Nil(t, err)
 	assert.Equal(t, csClear, userCs)
+
+	// Set a custom status for clearing it later
+	custom.Emoji = ":100"
+	custom.Text = "happy 100"
+
+	err = api.UpdateUserCustomStatus(user1.Id, custom)
+	assert.Nil(t, err)
+	userCs, err = th.App.GetCustomStatus(user1.Id)
+	assert.Nil(t, err)
+	assert.Equal(t, custom, userCs)
+
+	// Let plugin set a status
+	pluginCs := &model.CustomStatus{
+		Emoji:        ":calendar:",
+		Text:         "In meeting",
+		SetByProduct: true,
+	}
+	err = api.UpdateUserCustomStatus(user1.Id, pluginCs)
+	assert.Nil(t, err)
+	userCs, err = th.App.GetCustomStatus(user1.Id)
+	assert.Nil(t, err)
+	assert.Equal(t, pluginCs, userCs)
```

Comment:
With the above change applied

```suggestion
	assert.Equal(t, pluginCs, custom)
```

Reply:
because of https://github.com/mattermost/mattermost-server/pull/20851/files/96e78045b94da467f25dc34aab7fd998343c66b6#r996069510